src/wallet/wallet.cpp

```diff
@@ -2575,17 +2556,21 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             ++it;
     }
 
+    // form groups from remaining coins; note that preset coins will not
+    // automatically have their associated (same address) coins included
+    std::vector<OutputGroup> groups = group_outputs(vCoins, !(coinControl && coinControl->m_avoid_partial_spends > -1 ? coinControl->m_avoid_partial_spends : gArgs.GetBoolArg("-avoidpartialspends", DEFAULT_AVOIDPARTIALSPENDS)));
```

Comment:
I think `coinControl->m_avoid_partial_spends` should be a boolean, and `coinControl` should set its value based on the arg `avoidpartialspends` just after being created. (At worse, just before this call)

This will make the code easier to reason about.

Reply:
Unfortunately that means either adding a coincontrol.cpp file and moving the constructor there, or adding a dependency on the `gArg` thing in `coincontrol.h`, which seems a bit overkill just for this. I'm very open to suggestions though; the short thing is an undesired hack